### DISCUSSION TITLE
Add `onFocus` hook to `pane` in layout

### DISCUSSION THREAD

**Discussion by alex35mil** [zellij-org/zellij#2062] [Ideas]
My primary use case would be ensuring I'm in locked mode when I focus the pane with my editor.

```kdl
tab name="editor" focus=true {
  pane {
    borderless=true 
    onFocus={ SwitchToMode "Locked"; }
  }
}
```

**Comment by imsnif** [comment]
I really like this idea (only for consistency, better name it `on_focus`) - would you like to open an issue about it?

**Reply by alex35mil** [reply]
Snake case, of cource :) Creating an issue.
